File: Python/1963. Minimum Number of Swaps to Make the String Balanced.py
class Solution:
    def minSwaps(self, s: str) -> int:
        temp = []

        for i in s:
            if i == "]" and temp and temp[-1] == "[": temp.pop()
            else: temp.append(i)

        return math.ceil(Counter(temp)["["] / 2)

File: Java/1963. Minimum Number of Swaps to Make the String Balanced.java
class Solution {
    public int minSwaps(String s) {
//         Stack<Character> temp = new Stack<>();

//         for(char i: s.toCharArray()){
//             if(i == ']' && temp.size() > 0 && temp.lastElement() == '[') temp.pop();
//             else temp.push(i);
//         }

//         return (temp.size() + 2) / 4;

        int ans = 0;
        for(char i: s.toCharArray()){
            if(i == '[') ans++;
            else if(ans > 0) ans--;
        }

        return (ans + 1) / 2;
    }
}
